### DISCUSSION TITLE
Build issue - Could not find Boost, missing: iostreams program_options system serialization

### DISCUSSION THREAD

**Discussion by nimafo** [cdcseacave/openMVS#871] [Q&A]
Hi all, 
to build openMVS I run the following line of code:

`cmake  -DCMAKE_TOOLCHAIN_FILE=C:\vcpkg\scripts\buildsystems\vcpkg.cmake -DVCPKG_TARGET_TRIPLET=x64-windows -DBoost_INCLUDE_DIR='C:\boost_1_78_0' -DEIGEN3_INCLUDE_DIR="vcpkg install eigen3" -DVCG_ROOT="..\VCG" -S ..\src -G "Visual Studio 16 2019" ..`

This results in the following error:

> Could NOT find Boost (missing: iostreams program_options system
>   serialization) (found version "1.78.0")

in the CMakeLists.txt I see the following section: 

> FIND_PACKAGE**(Boost COMPONENTS iostreams program_options system serialization REQUIRED)
> if(Boost_FOUND)**
> 	LIST(APPEND OpenMVS_EXTRA_INCLUDES ${Boost_INCLUDE_DIRS})
> 	INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIRS})
> 	LIST(APPEND OpenMVS_DEFINITIONS -D_USE_BOOST)
> 	ADD_DEFINITIONS(${Boost_DEFINITIONS})
> 	LINK_DIRECTORIES(${Boost_LIBRARY_DIRS})
> 	SET(_USE_BOOST TRUE)
> endif()

Do you how to fix this issue? 
Thanks in advance!

**Comment by lrineau** [comment]
Try replace `-DBoost_INCLUDE_DIR='C:\boost_1_78_0'` by `-DBOOST_ROOT='C:\boost_1_78_0'`. The reason is that openMVS requires not only the headers of Boost, but also a few pre-compiled libraries (see the argument of `FIND_PACKAGE`, after `COMPONENTS`).

**Comment by nimafo** [comment]
Thank you @lrineau  for your response,
but it did not work even with `-DBOOST_ROOT='C:\boost_1_78_0'` and lots of other trials. 
I built it on WSL ubuntu in the end, as demonstrated [here](https://www.youtube.com/watch?v=B_2Y-aMp7Fw). and it seems to be working.
